### PR TITLE
Uncover new apps marketplace

### PR DESCRIPTION
**Pull request by orzechdev**
I want to merge this change because it uncovers new apps marketplace.

⚠️ First requires merge https://github.com/saleor/saleor-dashboard/pull/2975.

Closes https://github.com/saleor/saleor-app-marketplace/issues/146.

Removal of old apps code will be made in another PR.

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
MARKETPLACE_URL=https://apps.saleor.io
APPS_MARKETPLACE_API_URI=https://marketplace-gray.vercel.app/api/v2/saleor-apps
SALEOR_APPS_ENDPOINT=https://apps.saleor.io/api/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation
22. [ ] variants
23. [ ] payments

CONTAINERS=1

### CODE REVIEW

File: .github/workflows/deploy-demo-staging.yaml [hunk 1]
```diff
@@ -20,13 +20,7 @@ jobs:
...
       SENTRY_DSN: ${{ secrets.SENTRY_DSN }}
       SENTRY_AUTH_TOKEN: ${{ secrets.SENTRY_AUTH_TOKEN }}
-      # Disable preview when new Marketplace features are in progress.
-      # If marketplace is done till next release this can be easily enabled
-      # MARKETPLACE_URL: "https://apps.saleor.io/"
-      # SALEOR_APPS_PAGE_PATH=saleor-apps
-      # SALEOR_APPS_JSON_PATH=api/saleor-apps
-      # APP_TEMPLATE_GALLERY_PATH=template-gallery
-      # APPS_MARKETPLACE_API_URI: "https://apps.saleor.io/api/v2/saleor-apps"
+      APPS_MARKETPLACE_API_URI: "https://apps.saleor.io/api/v2/saleor-apps"
```
Comment (lkostrowski) [inline .github/workflows/deploy-demo-staging.yaml:23]: Staging -> `https://marketplace-gray.vercel.app/api/v2/saleor-apps`

File: .github/workflows/deploy-staging.yaml [hunk 2]
```diff
@@ -22,13 +22,7 @@ jobs:
...
       SENTRY_DSN: ${{ secrets.SENTRY_DSN }}
       SENTRY_AUTH_TOKEN: ${{ secrets.SENTRY_AUTH_TOKEN }}
-      # Disable preview when new Marketplace features are in progress.
-      # If marketplace is done till next release this can be easily enabled
-      # MARKETPLACE_URL: "https://apps.saleor.io/"
-      # SALEOR_APPS_PAGE_PATH=saleor-apps
-      # SALEOR_APPS_JSON_PATH=api/saleor-apps
-      # APP_TEMPLATE_GALLERY_PATH=template-gallery
-      # APPS_MARKETPLACE_API_URI: "https://marketplace-gray.vercel.app/api/v2/saleor-apps"
+      APPS_MARKETPLACE_API_URI: "https://apps.saleor.io/api/v2/saleor-apps"
```
Comment (lkostrowski) [inline .github/workflows/deploy-staging.yaml:25]: Staging -> `https://marketplace-gray.vercel.app/api/v2/saleor-apps`

**Review by lkostrowski — CHANGES_REQUESTED** [review]

**Review by lkostrowski — CHANGES_REQUESTED** [review]
Lets wait for https://github.com/saleor/saleor-dashboard/pull/2975

**Review by poulch — CHANGES_REQUESTED** [review]
LGTM, are you going to remove old apps dir ?

**Review by poulch — APPROVED** [review]

**Comment by lkostrowski** [comment]
> LGTM, are you going to remove old apps dir ?

@poulch it will be removed in another PR, we don't want to delay release with a big code diff in the same PR

**Review by NyanKiyoshi — APPROVED** [review]
CIs: LGTM

**Review by lkostrowski — APPROVED** [review]

**Pull request merged**